scripts/hid_configurator/NrfHidDevice.py

```diff
@@ -87,7 +98,63 @@ def parse_response(response_raw):
         else:
             event_data = data[:data_len]
 
-        return Response(rcpt, event_id, status, event_data)
+        return (rcpt, event_id, status, event_data)
+
+    @staticmethod
+    def exchange_feature_report(dev, recipient, event_id, status, event_data,
+                                poll_interval=POLL_INTERVAL_DEFAULT):
+        data = NrfHidTransport._create_feature_report(recipient, event_id, status, event_data)
+
+        try:
+            dev.send_feature_report(data)
+        except Exception as e:
+            logging.error('Send feature report problem: {}'.format(e))
+            return False, None
+
+        for _ in range(POLL_RETRY_COUNT):
+            time.sleep(poll_interval)
+
+            try:
+                response_raw = dev.get_feature_report(REPORT_ID, REPORT_SIZE)
+            except Exception as e:
+                logging.error('Get feature report problem: {}'.format(e))
+                rsp_status = ConfigStatus.FAULT
+                break
+
+            rsp = NrfHidTransport._parse_response(response_raw)
+            if rsp is None:
+                logging.error('Parsing response failed')
+                rsp_status = ConfigStatus.FAULT
+                break
+
+            (rsp_recipient, rsp_event_id, rsp_status, rsp_event_data) = rsp
```

Comment:
I don't think this is better then having a Response class. Why class removed?

Reply:
I wanted to separate HID feature report transport from NrfHidDevice.

The Response class was used only in one function (the one that exchanged feature reports). I think it was unnecessary and I wanted to handle the response in the same way as we handle requests (without separate class), just parsing the data using struct.